packages/core/src/sso/SamlConnector/utils.ts

```diff
@@ -1,34 +1,34 @@
 import * as validator from '@authenio/samlify-node-xmllint';
 import { ConnectorError, ConnectorErrorCodes } from '@logto/connector-kit';
-import { type Optional, conditional } from '@silverhand/essentials';
+import { type Optional, conditional, appendPath } from '@silverhand/essentials';
 import { got } from 'got';
 import * as saml from 'samlify';
 import { z } from 'zod';
 
+import { ssoPath } from '#src/routes/interaction/const.js';
+
 import {
-  samlMetadataGuard,
-  type SamlMetadata,
   type SamlConnectorConfig,
-  type SamlConfig,
   defaultAttributeMapping,
   type CustomizableAttributeMap,
   type AttributeMap,
   extendedSocialUserInfoGuard,
   type ExtendedSocialUserInfo,
+  type SamlIdentityProviderMetadata,
+  samlIdentityProviderMetadataGuard,
 } from '../types/saml.js';
 
 type ESamlHttpRequest = Parameters<saml.ServiceProviderInstance['parseLoginResponse']>[2];
 
 /**
- * Parse XML-format raw SAML metadata and return the parsed SAML metadata.
+ * Return the parsed SAML metadata using SAML identity provider initiated with SAML metadata.
  *
- * @param xml Raw SAML metadata in XML format.
- * @returns The parsed SAML metadata.
+ * @param idP SAML identity provider instance.
```

Comment:
IdP

Reply:
Input parameter should be camel case instead of pascal case?

Reply:
o yes you are right